pkg/controller/atlasproject/project_settings.go

```diff
@@ -0,0 +1,95 @@
+package atlasproject
+
+import (
+	"context"
+	"reflect"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func ensureProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) (result workflow.Result) {
+	if result = syncProjectSettings(ctx, projectID, project); !result.IsOk() {
+		ctx.SetConditionFromResult(status.ProjectSettingsReadyType, result)
+		return result
+	}
+
+	if project.Spec.Settings == nil {
+		ctx.UnsetCondition(status.ProjectSettingsReadyType)
+		return workflow.OK()
+	}
+
+	ctx.SetConditionTrue(status.ProjectSettingsReadyType)
+	return workflow.OK()
+}
+
+func syncProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) workflow.Result {
+	spec := project.Spec.Settings
+
+	atlas, err := fetchSettings(ctx, projectID)
+	if err != nil {
+		return workflow.Terminate(workflow.ProjectSettingsReady, err.Error())
+	}
+
+	if !areSettingsInSync(atlas, spec) {
+		if err := patchSettings(ctx, projectID, spec); err != nil {
+			return workflow.Terminate(workflow.ProjectSettingsReady, err.Error())
+		}
+	}
+
+	return workflow.OK()
+}
+
+func areSettingsInSync(atlas, spec *v1.ProjectSettings) bool {
```

Comment:
Maybe its better to use `isOneContainedInOther` func here

Reply:
I understand what you mean but I intentionally left it as is. I think having something like `isOneContainedInOther` on top level might be confusing and it reads better with `areSettingsInSync`. It's also possible to put the content of `isOneContainedInOther` into `areSettingsInSync` and remove the first func entirely but I'm also not a fan of this approach. This way the function is kind of like a in-code-comment and it explains the sync func and what it actually does.

Also lot's of the latest "project features" have a certain similarities in the code, including `..InSync` func, and I want to keep these similarities :)

Reply:
Yes, my original idea was to move the body of the contains function into this function. And add a comment to that function, which explains the logic of comparing structures.

Reply:
I think my version is better as we basically get the same readability but without any comments :)

Let's keep as is 😁